data_write/data_write.py

```diff
@@ -788,13 +788,32 @@ def find_expectation_value(x, parameters, field_name):
                 # so we get median of all sequences.
                 averaging_method = parameters['averaging_method']
                 array_2d = np.array(x, dtype=np.complex64)
+                num_beams, num_ranges, num_lags = parameters['correlation_dimensions']
+
+                # First range offset in samples
+                sample_off = parameters['first_range_rtt'] * 1e-6 * parameters['rx_sample_rate']
+                sample_off = np.int32(sample_off)
+
+                # Find sample number which corresponds with second pulse in sequence
+                tau_in_samples = parameters['tau_spacing'] * 1e-6 * parameters['rx_sample_rate']
+                second_pulse_sample_num = np.int32(tau_in_samples) * parameters['pulses'][1] - sample_off - 1
+
+                # Average the data
                 if averaging_method == 'mean':
                     array_expectation_value = np.mean(array_2d, axis=0)
                 elif averaging_method == 'median':
-                    array_expectation_value = np.median(array_2d, axis=0)
+                    array_expectation_value = np.median(np.real(array_2d), axis=0) +\
```

Comment:
I'm starting to wonder what exactly the median of complex numbers actually means... maybe we should be taking the median of the magnitude? I'm not sure

Reply:
I asked Pasha and he said to take the median of real and imag separately. I think maybe they're separate random variables ?